process/Style-guide.md

```diff
@@ -0,0 +1,250 @@
+      storage_writer (Optional[StorageWriter]): the storage writer must be open, 
+          and must not be closed. If no storage_writer is provided, a new writer 
+          will be created.
+    """
+```
+
+Make sure your arguments descriptions include:
+
+1. The argument(s) type(s);
+2. In case of [standard types](https://docs.python.org/3.3/library/stdtypes.html) a description of their format. Note that we use the Python 3 standard types;
+3. Description of the meaning of the argument. In other words how the argument is used by the function (or method). If the description exceeds the line limit, indent the next line with 4 spaces.
+
+The meaning can be left out if the function has only a couple arguments and how the arguments are used is obvious from the description as in the example of `AddAnalysisReport`.
+
+A few other tips:
+
+### Compound types
+
+If a function deals with a compound type (list, dict), document it like so:
+```
+Args:
+  constraints (dict[str, Filter]): constraint name mapped to the filter that implements the constraint.
+
+Returns:
+  list[BaseParser]: all relevant parsers.
+```
+
+### Multiple acceptable types
+
+If you need to specify multiple types, use a pipe to separate them. For example:
+
+```
+Args:
+  path (str|Path): path to tag file.
+```
+
+### Multiple return types
+
+Python simulates multiple arguments being returned by implicitly returning a tuple. Document like so:
+
+```
+...
+Returns:
+  tuple: containing:
+     
+    str: parser name
+    BaseParser: next parser parser
+""""
+return name, parser
+```
+
+### Special arguments
+
+Arguments like `cls`, `self`, `*args`, `**kwargs` should not be explicitly named in the `Args:` section.
+
+```
+  def CopyToIsoFormat(cls, timestamp, timezone=pytz.UTC, raise_error=False):
+    """Copies the timestamp to an ISO 8601 formatted string.
+
+    Args:
+      timestamp (int): number of micro seconds since January 1, 1970, 00:00:00 UTC.
+      timezone (Optional[pytz.timezone]): the result string will be expressed in this timezone.
+      raise_error (Optional[bool]): False if OverflowError should be caught when timestamp is out of bounds.
+
+    Returns:
+      str: ISO 8601 formatted date and time.
+    """
+```
+
+### Class attributes
+
+In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html)  sort class attribute alphabetically by name.
+
+```
+class SampleClass(object):
+  """Summary of class here.
+
+  Attributes:
+      eggs (int): number of eggs we have laid.
+      likes_spam (bool): whether we like SPAM or not.
+  """
+```
+
+### Constructors
+
+In addition to the [Google Python Style Guide](https://google.github.io/styleguide/pyguide.html) sort instance attribute alphabetically by name inside the constructor (`__init__`).
+
+```
+class SampleClass(object):
+  """Summary of class here."""
+
+  def __init__(self):
+    """Summary of method here."""
+    self.__private_attribute = None
```

Comment:
Is `self.__private_attribute` still accurate? Or should we change this? I don't think we use private attributes.

Reply:
Yeah, we don't use these. Removed these.